### PR TITLE
Add D* Producer for D* analysis.

### PR DESCRIPTION
**Pull request by geonmo**
- This producer makes D0 and D\* candidates for D\* analysis.
- We will also study to collect "right"  second pion meson while D0+ pion+- -> D\* +-

d0
dstar

### CODE REVIEW

**Comment by jshlee** [comment]
how big is this new collection?

**Comment by geonmo** [comment]
``` bash
catElectrons_catElectrons__CAT. 4240.5 3552.7
catMuons_catMuons__CAT. 3298.9 3290.8
catSecVertexs_catDstars_D0_CAT. 55152 7260.8
catSecVertexs_catDstars_Dstar_CAT. 5634 2968.6
```

Acutally, I can reduce the size about D0 using mass cut.
Dstar has a mass window about 50MeV

File: CatProducer/plugins/CATDStarProducer.cc [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    public:
+      explicit CATDStarProducer(const edm::ParameterSet & iConfig);
+      virtual ~CATDStarProducer() { }
+
+      void produce(edm::Event & iEvent, const edm::EventSetup & iSetup) override;
+
+    private:
+
+      vector<cat::SecVertex> *D0_Out_;
+      vector<cat::SecVertex> *Dstar_Out_;
```
Comment (jshlee) [inline CatProducer/plugins/CATDStarProducer.cc]: do these need to be global variables?

File: CatProducer/plugins/CATDStarProducer.cc [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  std::vector< std::vector<const reco::Candidate*> > jetsDaughters;
+
+  for (const pat::Jet & aPatJet : *jetHandle){
+    std::vector<const reco::Candidate*> jetDaughters;
+    int nDaughter = aPatJet.numberOfDaughters();
+    if ( nDaughter < 3 ) continue;
+    for( int idx = 0 ; idx < nDaughter ; idx++) {
+      if ( nDaughter > maxNumPFCand_ ) break;
+      jetDaughters.push_back( aPatJet.daughter(idx) );
+    }
```
Comment (jshlee) [inline CatProducer/plugins/CATDStarProducer.cc]: instead of making jetDaughters
can't you just do the D0Cand making here?

**Comment by jshlee** [comment]
you want this merged??

**Pull request merged**